test/unit/logger/CMakeLists.txt

```diff
@@ -3,4 +3,462 @@
 
 add_unit_test(logger
     logger.cpp
+)
+
+set(TEST_TARGET_NAME test-logger_env_var)
+add_executable(${TEST_TARGET_NAME}
+    env_var.cpp
+)
+target_link_libraries(${TEST_TARGET_NAME}
+    PRIVATE
+    ${PROJECT_NAME}::common
+    GTest::gtest_main
+)
+
+set(OUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/logger_test.log)
+
+function(add_logger_env_var_logfile_match_test name env_var test_case match_file)
+    set(full_name unit-logger_env_var_${name})
+
+    if(EXISTS ${OUT_FILE})
+        file(REMOVE ${OUT_FILE})
+    endif()
+
+    add_test(NAME ${full_name}
+        COMMAND ${TEST_TARGET_NAME}
+    )
+
+    add_test(NAME ${full_name}-validate-output
```

Comment:
Are you guaranteed that these tests are going to run in order? Why not just run the test in the match script?